src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/HttpClientJsonExtensions.Get.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Json
+{
+    /// <summary>
+    /// Contains the extensions methods for using JSON as the content-type in HttpClient.
+    /// </summary>
+    public static partial class HttpClientJsonExtensions
+    {
+        public static Task<object?> GetFromJsonAsync(this HttpClient client, string requestUri, Type type, JsonSerializerOptions? options = null, CancellationToken cancellationToken = default)
+        {
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            Task<HttpResponseMessage> taskResponse = client.GetAsync(requestUri, cancellationToken);
```

Comment:
Such calls to GetAsync like this are going to cause HttpClient to fully buffer the output.  Is that desirable?  I'd have assumed we'd want to use ResponseHeadersRead and let the deserializer read the response stream directly rather than first buffering all of the response data and then handing that off to the serializer.